### PR TITLE
Fix a KeyError crash that occurs when `-v` is passed

### PR DESCRIPTION
**Pull request by kurtmckee**
[sc-12117]

### CODE REVIEW

**Review by urimandujano — COMMENTED** [review]
Tried this out locally and this does patch that bug up. It's too bad that the `GlobusHTTPResponse` object doesn't expose the underlying request data anymore -- I can't remember if that was a private interface anymore. 

I added @rudyardrichter as a reviwer to get his perspective on the SDK usage just in case there's another way to accomplish this.

File: globus_automate_client/cli/helpers.py [hunk 1]
```diff
@@ -37,8 +38,15 @@
...
-        response_status_code = result.data["status_code"]
+        if isinstance(result._response, requests.Response):
+            base_request = result._response.request
+        elif (
+            result._wrapped and isinstance(result._wrapped._response, requests.Response)
+        ):
+            base_request = result._wrapped._response.request
+        else:
+            return "HTTP details are unavailable"
+        response_status_code = result.http_status
```
Comment (rudyardrichter) [inline globus_automate_client/cli/helpers.py:49]: Nice, I was thinking I might show up here and be able to suggest swapping to `http_status` or something else to that effect. I think this change looks good. Would it be helpful to think about a "debug output" mode on `GlobusHTTPResponse` for cases like this? I feel like `str(response)` being just the JSON could maybe be more helpful, and do something more like this function.

**Comment by rudyardrichter** [comment]
@urimandujano I don't think this is missing out on any SDK tricks or anything, but we could totally add something to this effect on `GlobusHTTPResponse`.

File: globus_automate_client/cli/helpers.py (hunk 1 above)
Comment (kurtmckee) [reply globus_automate_client/cli/helpers.py:49]: Thanks! Is the "debug output" something to change in this repo or in the SDK? If it's in the SDK I'd like to pass that feedback to the SDK team.

Comment (rudyardrichter) [reply globus_automate_client/cli/helpers.py:49]: Yeah, that's what I was getting at. I'll float that in our next SDK catch-up.

**Review by rudyardrichter — APPROVED** [review]

**Comment by urimandujano** [comment]
Thanks for the input here @rudyardrichter. I know accessing "private" attributes is a quick way to break things but it's super convenient for consistently parsing API responses at the CLI level. Adding a standard interface into the error/response details would help streamline some of this code. Let us know how that conversation goes!

**Review by urimandujano — APPROVED** [review]

File: globus_automate_client/cli/helpers.py (hunk 1 above)
Comment (kurtmckee) [reply globus_automate_client/cli/helpers.py:49]: Okay, great. I'm marking this as 'resolved' but please re-open as needed.

**Pull request merged**